terror_test/terror_test.go

```diff
@@ -81,85 +76,96 @@ func (s *testTErrorSuite) TestTraceAndLocation(c *C) {
 
 	var e6 error
 
-	c.Assert(errors.ErrorEqual(nil, nil), IsTrue)
-	c.Assert(errors.ErrorNotEqual(e1, e6), IsTrue)
+	s.True(errors.ErrorEqual(nil, nil))
+	s.True(errors.ErrorNotEqual(e1, e6))
 }
 
-func (s *testTErrorSuite) TestNewError(c *C) {
+func (s *TErrorTestSuite) TestNewError() {
 	today := time.Now().Weekday().String()
 	err := predefinedTextualErr.GenWithStackByArgs(today)
-	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[executor:ExecutorAbsent]executor is taking vacation at "+today)
+	s.NotNil(err)
+	s.Equal("[executor:ExecutorAbsent]executor is taking vacation at " + today, err.Error())
 }
 
-func (s *testTErrorSuite) TestRFCCode(c *C) {
+func (s *TErrorTestSuite) TestRFCCode() {
 	c1err1 := errors.Normalize("nothing", errors.RFCCodeText("TestErr1:Err1"))
 	c2err2 := errors.Normalize("nothing", errors.RFCCodeText("TestErr2:Err2"))
-	c.Assert(c1err1.RFCCode(), Equals, errors.RFCErrorCode("TestErr1:Err1"))
-	c.Assert(c2err2.RFCCode(), Equals, errors.RFCErrorCode("TestErr2:Err2"))
+	s.Equal(errors.RFCErrorCode("TestErr1:Err1"), c1err1.RFCCode())
+	s.Equal(errors.RFCErrorCode("TestErr2:Err2"), c2err2.RFCCode())
+
 	berr := errors.Normalize("nothing", errors.RFCCodeText("Blank:B1"))
-	c.Assert(berr.RFCCode(), Equals, errors.RFCErrorCode("Blank:B1"))
+	s.Equal(errors.RFCErrorCode("Blank:B1"), berr.RFCCode())
 }
 
-func (*testTErrorSuite) TestLineAndFile(c *C) {
+func (s *TErrorTestSuite) TestLineAndFile() {
 	err := predefinedTextualErr.GenWithStackByArgs("everyday")
 	_, f, l, _ := runtime.Caller(0)
 	terr, ok := errors.Cause(err).(*errors.Error)
-	c.Assert(ok, IsTrue)
+	s.True(ok)
+
 	file, line := terr.Location()
-	c.Assert(file, Equals, f)
-	c.Assert(line, Equals, l-1)
+	s.Equal(f, file)
+	s.Equal(l -1 , line)
 
 	err2 := predefinedTextualErr.GenWithStackByArgs("everyday and everywhere")
 	_, f2, l2, _ := runtime.Caller(0)
 	terr2, ok2 := errors.Cause(err2).(*errors.Error)
-	c.Assert(ok2, IsTrue)
+	s.True(ok2)
 	file2, line2 := terr2.Location()
-	c.Assert(file2, Equals, f2)
-	c.Assert(line2, Equals, l2-1)
+	s.Equal(f2, file2)
+	s.Equal(l2 - 1, line2)
 }
 
-func (*testTErrorSuite) TestWarpAndField(c *C) {
-	causeErr := errors.New("load from etcd meet error")
-	ErrGetLeader := errors.Normalize("fail to get leader", errors.RFCCodeText("member:ErrGetLeader"))
-	errWithWarpedCause := errors.Annotate(ErrGetLeader, causeErr.Error())
-	c.Assert(errWithWarpedCause.Error(), Equals, "load from etcd meet error: [member:ErrGetLeader]fail to get leader")
+func (s *TErrorTestSuite) TestWarpAndField() {
+	cause := errors.New("load from etcd meet error")
+	s.NotNil(cause)
+
+	err := errors.Normalize("fail to get leader", errors.RFCCodeText("member:ErrGetLeader"))
+	errWithCause := errors.Annotate(err, cause.Error())
+	s.NotNil(errWithCause)
+
+	s.Equal("load from etcd meet error: [member:ErrGetLeader]fail to get leader", errWithCause.Error())
 }
+
+
+func TestExampleTestSuite(t *testing.T) {
+	suite.Run(t, new(TErrorTestSuite))
+}
```

Comment:
no new line at end of file?

Reply:
I can attach one. `go fmt` doesn't complain it.